draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,453 @@
+
+Current version of `Experience APIs` in Logto are neither well-documented nor publicly available for the developers. This document aims to provide a comprehensive overview of the SIE model in Logto and introduce a new set of `Experience APIs` that allow developers to build a customized sign-in experience for their users.
+
+### 3.1 Terminology
+
+- **Sign-in experience(SIE)**: A user's experience when interacting with Logto platform. Including the front-end user interface and the back-end API calls.
+- **Experience APIs**: A set of APIs that allow end-users to interact with the Logto for identity verification and profile management.
+- **Interaction session**: A session that is initiated when a user starts interacting with the Logto platform. The session will be terminated once the user logs out or the session expires.
+- **Experience app**: The front-end application that provides the user interface for the user interaction flow.
+- **Social/Single Sign-On(SSO) identity**: A third-party identity provider that allows users to sign in to the Logto platform using their social media or enterprise SSO accounts.
+- **Identifier**: A piece of information that can be used to identify a user in the Logto system. For example, username, email, phone number, or social/SSO identity.
+- **Verification record**: A `verification record` is a group of security data can be used to verify a user's identity by such as password, verification code, or TOTP code.
+- **Verification code**: A one-time code sent to the user's email or phone number for identity verification.
+- **Verification id**: The unique identifier for a verification record. Generated by the Logto server when a verification flow is initiated. User need to provide the verification id along with additional security information to verify their identity. The verification id can also be used to track the verification record status within the interaction session.
+- **Multi-Factor Authentication(MFA)**: A security process that requires more than one verification factor to verify a user's identity.
+
+## 4. Definitions
+
+A complete user `Sign-in experience` in Logto consists of four key steps:
+
+1. **Identification**: The user needs to provide a unique `identifier` to the platform. This can be a username, email address, or third-party social identity. It will be used to uniquely identify the user in the system.
+2. **Verification**: The `verification` process is used to verify the user's identity by asking for additional security information. This can be a password, a one-time code, or a passkey. Upon the security settings multiple verification methods can be used.
+3. **Profile completion**: The user may be prompted to complete their `profile` and `mfa` settings during a interaction session. This can be a first-time user registration, a mandatory profile and `mfa` settings update for exiting users, or a additional third-party identity linking.
```

Comment:
why do we have "an additional third-party identity linking" in this step?

```suggestion
3. **Profile fulfillment**: During an interaction, the user may be prompted to complete their profile and MFA settings. This could be a requirement for user registration, a mandatory update for profile and MFA settings for existing users, and so on.
```